src/objc.d

```diff
@@ -217,7 +225,11 @@ extern (C++) void objc_FuncDeclaration_semantic_checkLinkage(FuncDeclaration fd)
 // MARK: init
 extern (C++) void objc_tryMain_dObjc()
 {
-    VersionCondition.addPredefinedGlobalIdent("D_ObjectiveC");
+    if (global.params.isOSX && global.params.is64bit)
```

Comment:
iOS will show up in LDC - eventually.  LDC won't use this function and instead will use this: https://github.com/ldc-developers/ldc/pull/1419/files#diff-800aa352b671d3466755572a9157a82aR102